02-toolbar/style.css

```diff
@@ -0,0 +1,232 @@
+* {
+    margin: 0;
+    padding: 0;
+    box-sizing: border-box;
+}
+
+html {
+    font-size: 10px;
+}
+body {
+    background-color: #223997;
+    height: 100vh;
+    position: relative;
+    font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
+    font-weight: 500;
+}
+
+.toolbar {
+    background-color: white;
+    width: 50rem;
+    height: 25rem;
+    position: relative;
+    top: 50vh;  
+    left: 50vw;
+    transform: translate(-50%, -50%);
+    border-radius: 0 0 10% 10%;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    flex-direction: column;
+}
+
+.toolbar__btns {
+    display: flex;
+    justify-content: space-around;
+    width: 100%;
+    height: 0vh;
+}
+
+.toolbar__btns .btn {
+    list-style: none;
+    position: relative;
+    width: 100px;
+    height: 50px;
+    /* clip-path: polygon(0 0, 100% 0, 100% 100%, 0 100%);
+    background: red; */   
+}
+
+.btn--label {
+    cursor: pointer;
+    position: relative;
+    top: 0; 
+    left: 0;
+    font-size: 1.9rem;
+    display: flex;
+}
+
+/* Icons positioning */
+
+.toolbar__btns .btn--one .icon::before, 
+.toolbar__btns .btn--two .icon::before {
+    left: 15px;
+}
+.toolbar__btns .btn--one .icon::after,
+.toolbar__btns .btn--two .icon::after {
+    left: 23px;
+}
+
+
+.toolbar__btns .btn--three .icon::before {
+    left: 7px;
+}
+.toolbar__btns .btn--three .icon::after {
+    left: 12px;
+}
+
+
+.toolbar__btns .btn--four .icon::before {
+    left: 22px;
+}
+.toolbar__btns .btn--four .icon::after {
+    left: 32px;
+}
+
+
+.input {
+    position: absolute;
+    top: 200rem;
+}
+
+.btn--text {
+    opacity: 0;
+    transition: all 500ms ease;
+}
+
+.input:checked ~ .btn--label .btn--text {
+    opacity: 1;
+    transform: translateY(-20px);
+}
+
+.input:checked ~ .btn--label {
+    color: #223997;
+}
+
+.btn--label.icon {
+    width: 8rem;
+    margin: auto;
+}
+
+
+/* TOP AND BOTTOM ICONS  */
+
+.icon::before, 
+.icon::after {
+    font-weight: 900;
+    font-family: 'Font Awesome 5 Free';
+    position: absolute;
+    top: 0; 
+    left: 0; 
+    color: #a5a3bd;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    /* vertical-align: -50%;  */
+}
+
+.icon::before {
+    font-size: 3rem;
+}
+
+.icon::after {
+    bottom: -150%;
+    font-size: .8rem;
+    transform: scale(0);
+    transition: all 300ms ease;
+    color: #223997;
+}
+
+.input:checked ~ .icon::before {
+    top: -5rem;
+    transform: translateY(-20px);
+    transition: all .2s linear;
```

Comment:
> 2. I want the icon::before to transition back down the same way it moves up after the button is unchecked.

For a transition to work in both ways the transition must be in the default selector. In this case instead of here, it should be at `.icon::before`.

It's also possible to have a `transition` in both selectors. For example:
```scss
.icon::before {
    transition: top 0.2s, transform 0.2s; // will take 0.2s to do the "leave" the animation
}

.input:checked ~.icon::before {
    transition: top 1s, transform 1s;  // will take 1s to do the "entry" animation
}
```

Reply:
I had a problem applying this on label text, what am I missing 

.btn--text {
    opacity: 0;
    width: 100%;
    height: 100%;
    text-align: center;
    transform: translateY(35px);
    transition: top 2s, transform 2s;
}


.input .btn--label .btn--text {
    opacity: 1;
    transform: translateY(30px);
}

.input:checked ~ .btn--label .btn--text {
    opacity: 1;
    transform: translateY(-20px);
}

Reply:
Hum... The 2nd selector doesn't make sense  `.input .btn--label .btn--text`. This does not exist. (ie, `.btn--text` is not a child of `.input`).

Reply:
Oh yeah I was going to delete that

Reply:
But the problem is still there, I can't get the label text (.btn--text) to transition back to it's starting position